not an StringLiteralExpException when using a constant in header name

When using a constant as header name I'm getting an exception, example: 

```    
import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION;

@Path("resources")
public class MyRestResource {

    @GET
    public Response get(@HeaderParam(AUTHORIZATION) String token) {

    }

}

```


Here is the stacktrace:

```
Caused by: java.lang.IllegalStateException: AUTHORIZATION is not an StringLiteralExpr
    at com.github.javaparser.ast.expr.Expression.asStringLiteralExpr (Expression.java:370)
    at com.sebastian_daschner.jaxrs_analyzer.analysis.javadoc.JavaDocParserVisitor.lambda$createMemberParamTag$6 (JavaDocParserVisitor.java:143)
    at java.util.stream.Collectors.lambda$toMap$58 (Collectors.java:1321)
    at java.util.stream.ReduceOps$3ReducingSink.accept (ReduceOps.java:169)
    at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:175)
    at java.util.ArrayList$ArrayListSpliterator.forEachRemaining (ArrayList.java:1382)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471
```

JaxRS analyzer 0.17-SNAPSHOT

--- a/src/main/java/com/sebastian_daschner/jaxrs_analyzer/analysis/javadoc/JavaDocParserVisitor.java
+++ b/src/main/java/com/sebastian_daschner/jaxrs_analyzer/analysis/javadoc/JavaDocParserVisitor.java
@@ -9,6 +9,8 @@ import com.github.javaparser.ast.body.Parameter;
 import com.github.javaparser.ast.comments.Comment;
 import com.github.javaparser.ast.expr.AnnotationExpr;
 import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
 import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
 import com.github.javaparser.javadoc.Javadoc;
 import com.github.javaparser.javadoc.JavadocBlockTag;
@@ -132,8 +134,19 @@ public class JavaDocParserVisitor extends VoidVisitorAdapter<Void> {
         Map<String, String> annotations = annotationStream
                 .filter(Expression::isSingleMemberAnnotationExpr)
                 .collect(Collectors.toMap(a -> a.getName().getIdentifier(),
-                a -> a.asSingleMemberAnnotationExpr().getMemberValue().asStringLiteralExpr().asString()));
+                a -> createMemberParamValue(a)));
         return new MemberParameterTag(javadocDescription.toText(), annotations);
+    }
+
+    private String createMemberParamValue(AnnotationExpr a) {
+        Expression memberValue = a.asSingleMemberAnnotationExpr().getMemberValue();
+        if(memberValue.getClass().isAssignableFrom(StringLiteralExpr.class)) {
+            return memberValue.asStringLiteralExpr().asString();
+        } else if(memberValue.getClass().isAssignableFrom(NameExpr.class)) {
+            return memberValue.asNameExpr().getNameAsString();
+        }
+        throw new IllegalArgumentException(String.format("Javadoc param type (%s) not supported.",memberValue.toString()));
+
     }
 
     private Map<Integer, String> createResponseComments(Javadoc javadoc) {

--- a/src/test/java/com/sebastian_daschner/jaxrs_analyzer/analysis/ProjectAnalyzerTest.java
+++ b/src/test/java/com/sebastian_daschner/jaxrs_analyzer/analysis/ProjectAnalyzerTest.java
@@ -275,6 +275,12 @@ public class ProjectAnalyzerTest {
                 .andResponse(200, ResponseBuilder.withResponseBody(stringIdentifier).build()).build();
         addMethods(resources, "complex/anotherSubres/{name}", eleventhGet);
 
+        // complex/auth
+        ResourceMethod authGet = ResourceMethodBuilder.withMethod(HttpMethod.GET, "Creates an authorization endpoint.")
+                .andHeaderParam("Authorization", Types.STRING)
+                .andResponse(200, ResponseBuilder.withResponseBody(stringIdentifier).build()).build();
+        addMethods(resources, "complex/auth", authGet);
+
         // json_tests
         final TypeIdentifier firstIdentifier = TypeIdentifier.ofDynamic();
         properties = new HashMap<>();

--- a/src/test/jaxrs-test/com/sebastian_daschner/jaxrs_test/ComplexResources.java
+++ b/src/test/jaxrs-test/com/sebastian_daschner/jaxrs_test/ComplexResources.java
@@ -25,7 +25,10 @@ import javax.ws.rs.container.ResourceContext;
 import javax.ws.rs.core.Context;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
+import javax.ws.rs.HeaderParam;
 import java.util.ArrayList;
+import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION;
+
 
 /**
  * These sources are solely used for test purposes and not meant for deployment.
@@ -82,6 +85,17 @@ public class ComplexResources extends AbstractResources implements Resources {
     public SomeSubResource anotherSubresResource() {
         // just for testing, this would fail due to missing default constructor
         return rc.getResource(SubResources.class);
+    }
+
+    /**
+     * Creates an authorization endpoint.
+     *
+     * @param token auth token
+     */
+    @GET
+    @Path("auth")
+    public Response get(@HeaderParam(AUTHORIZATION) String token) {
+        return Response.ok("Authorized").build();
     }
 
 }

